[core] Single module snapshot build fails

**Affects PMD Version:**  7.19.0-SNAPSHOT

**Description:**
Both [Newcomers' Guide](https://docs.pmd-code.org/latest/pmd_devdocs_contributing_newcomers_guide.html) and [Building PMD General Info](https://docs.pmd-code.org/latest/pmd_devdocs_building_general.html) document how to build a single module instead of a full build. This fails for me on a freshly cloned repository.

**Build error:**
Using pmd-apex as example.
```
[ERROR] Failed to execute goal on project pmd-apex: Could not resolve dependencies for project net.sourceforge.pmd:pmd-apex:jar:7.19.0-SNAPSHOT
[ERROR] dependency: net.sourceforge.pmd:pmd-core:jar:7.19.0-SNAPSHOT (compile)
[ERROR] 	Could not find artifact net.sourceforge.pmd:pmd-core:jar:7.19.0-SNAPSHOT
[ERROR] dependency: net.sourceforge.pmd:pmd-test:jar:7.19.0-SNAPSHOT (test)
[ERROR] 	Could not find artifact net.sourceforge.pmd:pmd-test:jar:7.19.0-SNAPSHOT
[ERROR] dependency: net.sourceforge.pmd:pmd-lang-test:jar:7.19.0-SNAPSHOT (test)
[ERROR] 	Could not find artifact net.sourceforge.pmd:pmd-lang-test:jar:7.19.0-SNAPSHOT
[ERROR] 
```

**Steps to reproduce:**
1. select a snapshot commit (single module builds on a release tag works)
2. `./mvnw clean verify -DskipTests` (full build succeeds)
3. `./mvnw verify -pl pmd-apex -DskipTests` (single module fails)

**Workaround:**
`./mvnw verify -pl pmd-apex -am -DskipTests` works for me.
So maybe just a parameter missing in the documentation.

--- a/docs/pages/pmd/devdocs/building/building_general.md
+++ b/docs/pages/pmd/devdocs/building/building_general.md
@@ -126,6 +126,13 @@ See also <https://central.sonatype.org/publish/publish-portal-snapshots/>.
   ```shell
   ./mvnw verify -pl pmd-apex
   ```
+* If you are currently on a snapshot commit (as opposed to a release commit) you need snapshot modules as dependencies
+  for the module you want to build. You can either build them yourself (line 1) or you can activate the snapshot
+  repository (line 2) for your build:
+  ```shell
+  ./mvnw verify -pl pmd-apex -am
+  ./mvnw verify -pl pmd-apex -Pcentral-portal-snapshots
+  ```
   **Caveats:** We have some integration tests, that run only after all modules have been built. You could
   break these without noticing.  
   **Note:** In our CI (via GitHub Actions) we always build the complete project.

--- a/docs/pages/pmd/devdocs/contributing/newcomers_guide.md
+++ b/docs/pages/pmd/devdocs/contributing/newcomers_guide.md
@@ -84,6 +84,17 @@ pull requests later for your fix.
     $ ./mvnw clean verify -pl pmd-apex
     ```
 
+6.  You might get an error about not being able to resolve a `snapshot` module:
+    ```shell
+    Could not resolve dependencies for project net.sourceforge.pmd:pmd-apex:jar:7.20.0-SNAPSHOT
+    ```
+    Then you are on snapshot commit and need snapshot dependencies for your module. You can activate building the
+    required dependencies yourself (line 1) or you can activate the snapshot repository (line 2) for your build:
+    ```shell
+    $ ./mvnw clean verify -pl pmd-apex -am
+    $ ./mvnw clean verify -pl pmd-apex -Pcentral-portal-snapshots
+    ```
+
     We recommend to read the documentation [Building PMD General Info](pmd_devdocs_building_general.html) and
     especially the IDE specific guides, such as [Building PMD with IntelliJ IDEA](pmd_devdocs_building_intellij.html).
     These pages explain how to prepare your local development environment in order to work on PMD.

